pkg/services/dashboards/service/dashboard_public.go

```diff
@@ -58,3 +61,31 @@ func (dr *DashboardServiceImpl) SavePublicDashboardConfig(ctx context.Context, d
 
 	return pdc, nil
 }
+
+func (dr *DashboardServiceImpl) BuildPublicDashboardMetricRequest(ctx context.Context, publicDashboardUid string, panelId int64) (dtos.MetricRequest, error) {
+	pdc, d, err := dr.dashboardStore.GetPublicDashboard(publicDashboardUid)
```

Comment:
these are pretty confusing variable names :)

Reply:
Fair enough. Let me fix that :)